### PR TITLE
check unique only for face-up position

### PR DESCRIPTION
**Pull request by salix5**
@DailyShana @mercury233 
Problem
[unique.zip](https://github.com/Fluorohydride/ygopro-core/files/4807459/unique.zip)
unique1.yrp
Turn 4
Now: マジカルシルクハット cannot special summon 宮廷のしきたり if another one is on the field.
Correct: マジカルシルクハット can special summon 宮廷のしきたり if another one is on the field.

unique2.yrp
Now: Face-down 炎の王 ナグルファー cannot return to field if another one is on the field.
Correct: Face-down 炎の王 ナグルファー can return to field if another one is on the field.

Reason
unique1.yrp
unique2.yrp
It should check uniqueness only when monsters enter the field in face-up position.

### CODE REVIEW

**Comment by DailyShana** [comment]
keep consistent with
https://github.com/Fluorohydride/ygopro-core/blob/887d70fa612888c23c7bccb8b8fcc23257841d93/card.cpp#L3153

**Comment by salix5** [comment]
Thank you.

**Comment by DailyShana** [comment]
it is also possible to special summon in position facedown by `EFFECT_SPSUMMON_PROC` with `EFFECT_FLAG_SPSUM_PARAM`. Would you like add this check in `filter_spsummon_procedure`

**Comment by salix5** [comment]
Done.
Moreover, is_can_be_special_summoned() will also check EFFECT_DEVINE_LIGHT now.

**Comment by DailyShana** [comment]
Thank you

File: card.cpp [hunk 1]
```diff
@@ -3150,7 +3150,9 @@ int32 card::is_can_be_special_summoned(effect* reason_effect, uint32 sumtype, ui
 	if((data.type & TYPE_PENDULUM) && current.location == LOCATION_EXTRA && (current.position & POS_FACEUP)
 		&& (sumtype == SUMMON_TYPE_FUSION || sumtype == SUMMON_TYPE_SYNCHRO || sumtype == SUMMON_TYPE_XYZ))
 		return FALSE;
-	if(((sumpos & POS_FACEDOWN) == 0) && pduel->game_field->check_unique_onfield(this, toplayer, LOCATION_MZONE))
+	if (sumpos & POS_FACEDOWN && pduel->game_field->is_player_affected_by_effect(sumplayer, EFFECT_DEVINE_LIGHT))
```
Comment (DailyShana) [inline card.cpp:3153]: a little format improve
```suggestion
	if((sumpos & POS_FACEDOWN) && pduel->game_field->is_player_affected_by_effect(sumplayer, EFFECT_DEVINE_LIGHT))
```

**Pull request merged**